### PR TITLE
Fairing for timing requests

### PR DESCRIPTION
**Pull request by marcelometal**
Addded X-Response-Time header

### CODE REVIEW

File: src/x_response_time.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        request.local_cache(|| TimerStart(Some(SystemTime::now())));
+    }
+
+    /// Adds a header to the response indicating how long the server took to
+    /// process the request.
+    fn on_response(&self, request: &Request, response: &mut Response) {
+        let start_time = request.local_cache(|| TimerStart(None));
+        if let Some(Ok(duration)) = start_time.0.map(|st| st.elapsed()) {
+            let ms = duration.as_secs() * 1000 + duration.subsec_millis() as u64;
+            response.set_raw_header("X-Response-Time", format!("{} ms", ms));
```
Comment (kov) [inline src/x_response_time.rs:39]: What about printing it as well? You can use info!() for that

File: src/x_response_time.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }
+    }
+}
+
+/// Request guard used to retrieve the start time of a request.
```
Comment (kov) [inline src/x_response_time.rs]: I _think_ we do not need anything from this line onwards. Unless you plan to use the timer on a route function.

Comment (marcelometal) [reply src/x_response_time.rs]: Ok! I will remove it.

File: src/x_response_time.rs (hunk 1 above)
Comment (marcelometal) [reply src/x_response_time.rs:39]: maybe both, because we can use the "X-Response-Time" header in frontend,

Comment (kov) [reply src/x_response_time.rs:39]: Yep!

**Pull request merged**